### PR TITLE
[EFR32] Clean up gn files for QRcode Lib

### PR DESCRIPTION
**Pull request by mkardous-silabs**
#### Problem
* QRCode lib was always added to the dependencies when it wasn't always necessar 
* Window app did not build when disabling QR Code

#### Change overview
* Move QRcode dependencies
* Fix Window app when not using the QR code

#### Testing
* Manual tests to validate no regressions where caused
* Manual tests with window-app to validate that it builds and works without the QR code

### CODE REVIEW

**Review by bzbarsky-apple — APPROVED** [review]

File: examples/ota-requestor-app/efr32/BUILD.gn [hunk 1]
```diff
@@ -126,6 +126,8 @@ efr32_executable("ota_requestor_app") {
   if (show_qr_code) {
     sources += [ "${examples_plat_dir}/display/lcd.c" ]
     defines += [ "DISPLAY_ENABLED" ]
+
+    deps += [ "${chip_root}/examples/common/QRCode" ]
```
Comment (jepenven-silabs) [inline examples/ota-requestor-app/efr32/BUILD.gn:130]: Forgot to removed it from the deps above

File: examples/platform/efr32/display/lcd.c [hunk 2]
```diff
@@ -23,7 +23,11 @@
 
 #include "dmd.h"
 #include "glib.h"
+
+#if !defined(QR_CODE_ENABLED) || QR_CODE_ENABLED
```
Comment (jepenven-silabs) [inline examples/platform/efr32/display/lcd.c:27]: Wrong check here. should be `#if defined(QR_CODE_ENABLED) && QR_CODE_ENABLED`

File: examples/platform/efr32/display/lcd.c [hunk 3]
```diff
@@ -32,10 +36,14 @@
 #define QR_CODE_BORDER_SIZE 0
 
 static GLIB_Context_t glibContext;
+#if !defined(QR_CODE_ENABLED) || QR_CODE_ENABLED
```
Comment (jepenven-silabs) [inline examples/platform/efr32/display/lcd.c:39]: Same here

File: examples/platform/efr32/display/lcd.c [hunk 4]
```diff
@@ -32,10 +36,14 @@
...
 static uint8_t qrCode[qrcodegen_BUFFER_LEN_FOR_VERSION(QR_CODE_VERSION)];
 static uint8_t workBuffer[qrcodegen_BUFFER_LEN_FOR_VERSION(QR_CODE_VERSION)];
+#endif // !defined(QR_CODE_ENABLED) || QR_CODE_ENABLED
 
+#if !defined(QR_CODE_ENABLED) || QR_CODE_ENABLED
```
Comment (jepenven-silabs) [inline examples/platform/efr32/display/lcd.c:44]: Same

File: examples/platform/efr32/display/lcd.c [hunk 5]
```diff
@@ -92,6 +100,7 @@ int LCD_update(void)
     return DMD_updateDisplay();
 }
 
+#if !defined(QR_CODE_ENABLED) || QR_CODE_ENABLED
```
Comment (jepenven-silabs) [inline examples/platform/efr32/display/lcd.c:103]: Same

File: examples/window-app/efr32/src/WindowAppImpl.cpp [hunk 6]
```diff
@@ -431,14 +433,16 @@ void WindowAppImpl::UpdateLCD()
             LcdPainter::Paint(type, static_cast<uint8_t>(lift.Value()), static_cast<uint8_t>(tilt.Value()), mIcon);
         }
     }
+#if !defined(QR_CODE_ENABLED) || QR_CODE_ENABLED
```
Comment (jepenven-silabs) [inline examples/window-app/efr32/src/WindowAppImpl.cpp:436]: Same

**Review by jepenven-silabs — CHANGES_REQUESTED** [review]

File: examples/ota-requestor-app/efr32/BUILD.gn (hunk 1 above)
Comment (mkardous-silabs) [reply examples/ota-requestor-app/efr32/BUILD.gn]: done

File: examples/platform/efr32/display/lcd.c (hunk 3 above)
Comment (mkardous-silabs) [reply examples/platform/efr32/display/lcd.c]: as per discussed, logic has been changed

**Review by jepenven-silabs — APPROVED** [review]

**Pull request merged**